tests/SerdeTestCases.php

```diff
@@ -1187,6 +1188,41 @@ public function datetime_fields_support_custom_output_format(): void
 
     }
 
+    #[Test]
+    public function unixtime_fields_support(): void
+    {
+        $s = new SerdeCommon(formatters: $this->formatters);
+
+        $timeString = '4 July 2022 14:22:22';
+        $zone = new \DateTimeZone('America/New_York');
+
+        $stamp = new \DateTime($timeString, $zone);
+        $stampImmutable = new \DateTimeImmutable($timeString, $zone);
+        $data = new UnixTimeExample(seconds: $stamp, milliseconds: $stampImmutable);
+
+        $serialized = $s->serialize($data, $this->format);
+
+        switch($serialized) {
+            case "seconds: 1656958942\nmilliseconds: 1656958942000\n": // yaml
+            case '{"seconds":1656958942,"milliseconds":1656958942000}': // json
+            case ['seconds' => 1656958942, 'milliseconds' => 1656958942000]: // array
+                self::assertTrue(true);
+                break;
+            default:
+                self::assertTrue(false);
+                break;
+        }
+
+        // Because some of the exported formats involve data loss,
+        // we don't actually expect the exact same thing back.
+        $expected = new UnixTimeExample($stamp, $stampImmutable);
+
+        $result = $s->deserialize($serialized, from: $this->format, to: UnixTimeExample::class);
```

Comment:
```suggestion
        $result = $s->deserialize($serialized, from: $this->format, to: $data::class);
```

(Newer tests do it this way, because I was dumb and didn't realize I could do it when writing the older tests. 😄  One of these days I should update the older ones.)